src/features/applicationView/berthSwitchOfferCard/BerthSwitchOfferCard.tsx

```diff
@@ -0,0 +1,166 @@
+import { BerthSwitchOfferDetails, PlaceProperty } from './types';
+import Property from '../../../common/property/Property';
+import styles from './berthSwitchOfferCard.module.scss';
+import { IconFence, IconPlug, IconStreetLight, IconWaterTap } from '../../../common/icons';
+import PlaceDetails from '../berthOfferCard/PlaceDetails';
+import Button from '../../../common/button/Button';
+import { SEND_BERTH_SWITCH_OFFER_MUTATION } from './mutations';
+import {
+  SEND_BERTH_SWITCH_OFFER,
+  SEND_BERTH_SWITCH_OFFERVariables as SEND_BERTH_SWITCH_OFFER_VARS,
+} from './__generated__/SEND_BERTH_SWITCH_OFFER';
+import hdsToast from '../../../common/toast/hdsToast';
+import StatusLabel from '../../../common/statusLabel/StatusLabel';
+import { OFFER_STATUS } from '../../../common/utils/constants';
+import { getProfileToken } from '../../../common/utils/auth';
+import Modal from '../../../common/modal/Modal';
+import SendSwitchOfferForm from './sendSwitchOfferForm/SendSwitchOfferForm';
+
+export interface BerthSwitchOfferCardProps {
+  className?: string;
+  refetchQueries: PureQueryOptions[] | string[];
+  switchOffer: BerthSwitchOfferDetails;
+}
+
+const BerthSwitchOfferCard = ({
+  className,
+  refetchQueries,
+  switchOffer: {
+    berthComment,
+    berthDepth,
+    berthIsAccessible,
+    berthLength,
+    berthMooringType,
+    berthNum,
+    berthWidth,
+    electricity,
+    gate,
+    harborName,
+    id,
+    lighting,
+    pierIdentifier,
+    status,
+    wasteCollection,
+    water,
+  },
+}: BerthSwitchOfferCardProps) => {
+  const { t } = useTranslation();
+
+  const [sendOfferModalOpen, setSendOfferModalOpen] = useState<boolean>(false);
+  const [sendBerthSwitchOfferMutation, { loading: isSubmitting }] = useMutation<
+    SEND_BERTH_SWITCH_OFFER,
+    SEND_BERTH_SWITCH_OFFER_VARS
+  >(SEND_BERTH_SWITCH_OFFER_MUTATION, {
+    refetchQueries: refetchQueries,
+  });
+  const sendBerthSwitchOffer = ({ dueDate }: { dueDate: string }) => {
+    sendBerthSwitchOfferMutation({
+      variables: {
+        input: {
+          offers: [id],
+          profileToken: getProfileToken(),
+          dueDate: dueDate,
+        },
+      },
+    }).then((res) => {
+      if (res.data?.sendBerthSwitchOffer?.failedOffers && res.data?.sendBerthSwitchOffer?.failedOffers.length > 0) {
+        hdsToast({
+          type: 'error',
+          labelText: 'toast.invoiceError.label',
+          text: 'toast.invoiceError.description',
+          translated: true,
+        });
+      } else if (!res.errors) {
+        hdsToast({
+          type: 'success',
+          labelText: 'toast.invoiceSent.label',
+          text: 'toast.invoiceSent.description',
+          translated: true,
+        });
+      }
+    });
+  };
+
+  const isNotNull = (property: boolean | null): property is boolean => property !== null;
+  const mapPlaceProperties = (placeProperties: PlaceProperty[]) =>
+    placeProperties.map(
+      ({ prop, key, icon }) =>
+        isNotNull(prop) && (
```

Comment:
```suggestion
       prop ?? (
```

Reply:
The logic is backwards, but thanks for the suggestion. Going with this:
```
        prop !== null && (
```